docs/src/tutorials/entry-tutorial.md

```diff
@@ -0,0 +1,1085 @@
+{{#include ../../../test/tla/bin-search/BinSearch1.tla:8:18}}
+```
+
+Importantly, the constants `INPUT_SEQ` and `INPUT_KEY` are prefixed with type
+annotations in the comments:
+
+ - `INPUT_SEQ` has the type `Seq(Int)`, that is, it is a sequence of integers, and
+ - `INPUT_KEY` has the type `Int`, that is, it is an integer.
+
+Recall that we wanted to specify signed and unsigned Java integers, which are
+32 bit long. *TLA+ is not tuned towards any computer architecture.* Its integers
+are always signed and can be arbitrary large (unbounded), like in school math.
+To model fixed bit-width integers, we introduce another constant `INT_WIDTH` of
+type `Int`:
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch1.tla:19:22}}
+```
+
+The benefit of defining the bit width as a parameter is that we can try
+our specification for various bit widths of integers: 4-bit, 8-bit, 16-bit, 32-bit,
+etc.  Similar to many programming languages, we introduce several constant
+definitions (`a^b` is `a` taken to the power of `b`):
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch1.tla:24:30}}
+```
+
+Note that these definitions do not constrain integers in any way. They are
+simply convenient names for the constants that we will need in the
+specification.
+
+To make sure that the new specification does not contain syntax errors or
+type errors, execute:
+
+```sh
+$ apalache-mc check BinSearch1.tla
+```
+
+TODO: add the link to BinSearch1.tla when we have it on github.
+
+## Step 2: Specifying the base case
+
+Copy `BinSearch1.tla` to `BinSearch2.tla` and open `BinSearch2.tla`.
+
+We start with the simplest possible case that occurs in `binarySearch`. Namely,
+we consider the case of `low > high`, that is, `binarySearch` is never entering
+the loop.
+
+**Introduce variables**. To do that, we have to finally introduce some
+variables. Obviously, we have to introduce variables `low` and `high`. This is
+how we do it:
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch2.tla:33:38}}
+```
+
+The variables `low` and `high` are called *state variables*. They are always
+present in a state of our state machine. That is, they are never introduced and
+never removed. Remember, TLA+ is not tuned towards any particular computer
+architecture and thus it does not even have the notion of an execution stack.
+You can think of `low` and `high` as being global variables. Yes, global
+variables are generally a bad smell in a program. However, when dealing with a
+specification, they are much easier to reason about than the execution stack.
+If you need to introduce local definitions, you will see how to do that later
+in this tutorial.
+
+Less obviously, we introduce two additional variables:
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch2.tla:39:44}}
+```
+
+The variable `isTerminated` indicates, whether our search has terminated. Why
+did we even have to introduce it? Because, in general, computer systems do not
```

Comment:
```suggestion
The variable `isTerminated` indicates whether our search has terminated. Why
do we even have to introduce it? Because, in general, computer systems do not
```